### PR TITLE
Reorganise files in resources folder

### PR DESCRIPTION
**Pull request by dalepotter**
This pull request contains various changes to the structure of the resources, as agreed during discussions with @hayfield on the design of the library.

### CODE REVIEW

File: iati/core/tests/test_resources.py [hunk 1]
```diff
@@ -43,49 +43,21 @@ def test_find_schema_paths(self):
...
             assert iati.core.resources.PATH_SCHEMAS_202 in path
 
-    @pytest.mark.parametrize('name,location', [
-        ('Name', None),
-        ('Name', 'embedded'),
-        ('Name', 'non-embedded'),
-        ('Name.xml', None),
-        ('Name.xml', 'embedded'),
-        ('Name.xml', 'non-embedded'),
+    @pytest.mark.parametrize('name', [
```
Comment (hayfield) [inline iati/core/tests/test_resources.py]: Now that `location` has been removed, duplicates could be removed from this list.

File: iati/core/resources.py [hunk 2]
```diff
@@ -133,26 +122,15 @@ def path_codelist(name, location='non-embedded', version=None):
...
         name = name[:-4]
 
-    if location == 'embedded':
-        return path_for_version(os.sep.join((PATH_CODELISTS_EMBEDDED, '{0}'.format(name) + FILE_CODELIST_EXTENSION)), version)
-    elif location == 'non-embedded':
-        return path_for_version(os.sep.join((PATH_CODELISTS_NON_EMBEDDED, '{0}'.format(name) + FILE_CODELIST_EXTENSION)), version)
-    else:
-        msg = "The location of a Codelist must be a string equal to either 'embedded' or 'non-embedded'"
-        iati.core.utilities.log_error(msg)
-        raise ValueError(msg)
```
Comment (hayfield) [inline iati/core/resources.py]: Probably worth checking use of `Raises` in docstrings in this file.

File: iati/core/default.py [hunk 3]
```diff
@@ -82,7 +82,8 @@ def codelists(version=None, bypass_cache=False):
     paths = iati.core.resources.find_all_codelist_paths()
 
     for path in paths:
-        name = path.split(os.sep).pop()[:-len(iati.core.resources.FILE_CODELIST_EXTENSION)]
+        head, tail = os.path.split(path)
```
Comment (hayfield) [inline iati/core/default.py]: `head` looks unused - could use `_`

Comment (hayfield) [inline iati/core/default.py]: `tail` could possibly be renamed to better indicate that it's the file name

Comment (dalepotter) [inline iati/core/default.py]: Resolved in e3b2246

File: iati/core/resources.py (hunk 2 above)
Comment (dalepotter) [inline iati/core/resources.py]: This is resolved in c6d40e2

**Pull request merged**